### PR TITLE
Executable tests on Sylius-Standard & remove duplicated configs

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
General note, I have not reviewed it in full yet: We need to have same structure as Symfony Standard whenever possible, even if this means duplication of some configs. I don't want to write a whole chapter in docs about why configs are splitted and included separately and answering questions: "where is configuration X, in Symfony it is right there!". I'd rather still live with a small duplication.

**Comment by pamil** [comment]
@pjedrzejewski I agree, but there are over 1500 lines of configuration compared to 100 lines of Symfony Standard, so we must think through what we can duplicate and what rather not.

**Comment by pjedrzejewski** [comment]
The Sylius defaults should not be duplicated of course and I believe they could stay in the Sylius[Shop/Admin/Core accordingly]Bundle, just like they are right now. In main config files I would leave the basic Symfony/Doctrine settings you find in all Symfony apps and anything that's cannot be overriden (fos_rest paths IIRC are an example)

File: app/AppKernel.php [hunk 1]
```diff
@@ -23,15 +24,153 @@ class AppKernel extends Kernel
...
+            new \Payum\Bundle\PayumBundle\PayumBundle(),
+            new \JMS\SerializerBundle\JMSSerializerBundle(),
+            new \JMS\TranslationBundle\JMSTranslationBundle(),
+            new \Stof\DoctrineExtensionsBundle\StofDoctrineExtensionsBundle(),
+            new \WhiteOctober\PagerfantaBundle\WhiteOctoberPagerfantaBundle(),
+
+            new \Doctrine\Bundle\MigrationsBundle\DoctrineMigrationsBundle(),
+            new \Doctrine\Bundle\FixturesBundle\DoctrineFixturesBundle(),
+            new \Sylius\Bundle\FixturesBundle\SyliusFixturesBundle(),
+            new \Sylius\Bundle\PayumBundle\SyliusPayumBundle(), // must be added after PayumBundle.
```
Comment (michalmarcinkowski) [inline app/AppKernel.php]: This comment doesn't make sense 😄

File: app/config/config.yml [hunk 2]
```diff
@@ -2,88 +2,54 @@
 # (c) Paweł Jędrzejewski
 
 imports:
-    - { resource: parameters.yml }
-    - { resource: security.yml }
-    - { resource: migrations.yml }
-    - { resource: @SyliusCoreBundle/Resources/config/app/main.yml }
-    - { resource: @SyliusAdminBundle/Resources/config/app/main.yml }
+    - { resource: "security.yml" }
+
```
Comment (michalmarcinkowski) [inline app/config/config.yml]: `parameters.yml` shouldn't be there also?

File: app/config/config_test.yml [hunk 3]
```diff
@@ -65,3 +38,29 @@ payum:
...
+    disable_delivery: true
+
+sylius_channel:
+    fake_channel_support: true
+
+sylius_payment:
+    gateways:
+        cc: Credit Card
+        test: Test
+        stripe: Stripe
```
Comment (michalmarcinkowski) [inline app/config/config_test.yml]: Paypal is missing.

File: src/Sylius/Bundle/AddressingBundle/Resources/config/config.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# This file is part of the Sylius package.
+# (c) Paweł Jędrzejewski
+
+jms_serializer:
+    metadata:
+        directories:
+            sylius-addressing:
+                namespace_prefix: "Sylius\\Component\\Addressing"
+                path: "@SyliusAddressingBundle/Resources/config/serializer"
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/config.yml]: Why do we need this file?

File: src/Sylius/Bundle/CoreBundle/Resources/config/config.yml [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+            timestampable: true
+            loggable: true
+            sortable: true
+
+twig:
+    debug: "%kernel.debug%"
+    strict_variables: "%kernel.debug%"
+    exception_controller: "FOS\\RestBundle\\Controller\\ExceptionController::showAction"
+
+winzou_state_machine:
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/config.yml]: IMO this should be extracted to a separate file.

**Comment by pjedrzejewski** [comment]
* I think we should separate app configurations from services/parameters. Not sure if current ``config/app`` directory is a good idea, but having one big ``config.yml`` next to ``services.xml``, ``form.xml``, etc. can be confusing.
* State machines should definitely be in separate files.
* Why removing base Kernel class?

File: src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml [hunk 6]
```diff
@@ -138,101 +39,122 @@ sylius_archetype:
...
-            variant:
+            subject: "%sylius.model.product.class%"
+            association:
                 classes:
-                    model: Sylius\Component\Core\Model\ProductVariant
-                    repository: Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductVariantRepository
-                    form:
-                        default: Sylius\Bundle\CoreBundle\Form\Type\ProductVariantType
+                    model: Sylius\Component\Product\Model\ProductAssociation
+                    interface: Sylius\Component\Product\Model\ProductAssociationInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml]: Not a concern of this PR, but this shouldn't live in `ProductBundle`?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml]: This way we would couple ProductBundle with AssociationBundle.

**Comment by pamil** [comment]
@pjedrzejewski:

 - Moved back these files to bundles' `config/app/*`
 - Extracted state machine configuration to bundles' `config/app/state_machine.yml`
 - The kernel class from CoreBundle was moved to `/app/`, because it also includes WebBundle / AdminBundle etc. and it's definitely not a property of CoreBundle

**Comment by pamil** [comment]
I think it's all for this iteration, if there aren't any more issues, it would be nice to get this PR merged so I can open the another one on Sylius-Standard.

File: src/Sylius/Behat/Resources/config/profiles/default.yml [hunk 7]
```diff
@@ -40,8 +40,8 @@ default:
 
         Behat\Symfony2Extension:
             kernel:
-                class: TestKernel
-                path: app/TestKernel.php
+                class: TestAppKernel
+                path: app/TestAppKernel.php
```
Comment (lchrusciel) [inline src/Sylius/Behat/Resources/config/profiles/default.yml]: Shouldn't path have `"`?

Comment (pamil) [inline src/Sylius/Behat/Resources/config/profiles/default.yml]: It's not required in this case.

File: src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml (hunk 6 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml]: `Sylius\Component\Product\Model\ProductAssociation` - it is already coupled :)

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/sylius.yml]: And probably with the bundle ([composer.json](https://github.com/michalmarcinkowski/Sylius/blob/f2ef70f73186b6ca73d658abde1d5310c8fcf270/src/Sylius/Bundle/ProductBundle/composer.json#L29-L29)).

File: app/migrations/Version20160523143611.php [hunk 8]
```diff
@@ -17,7 +8,7 @@
 /**
  * Auto-generated Migration: Please modify to your needs!
  */
-class Version20160516131553 extends AbstractMigration
+class Version20160523143611 extends AbstractMigration
```
Comment (michalmarcinkowski) [inline app/migrations/Version20160523143611.php]: Why this migration was not removed?

Comment (pamil) [inline app/migrations/Version20160523143611.php]: It's the first one made after v0.18 was tagged

**Comment by lchrusciel** [comment]
Can we setup test environment based on migration rather than doctrine schema generation?

**Comment by bendavies** [comment]
@lchrusciel I would say the travis build should do that too. That way, travis can run `doctrine:schema:validate` a check if any schema changes have not been accompanied with a migration. which has happened more than once.

**Comment by lchrusciel** [comment]
@bendavies this is exactly what I meant :D We should ensure, that this won't happen again.

**Comment by bendavies** [comment]
@lchrusciel ah right! i thought you meant the demo env or something.

File: src/Sylius/Bundle/ShippingBundle/Resources/config/app/state_machine.yml [hunk 9]
```diff
@@ -0,0 +1,88 @@
+# This file is part of the Sylius package.
+# (c) Paweł Jędrzejewski
+
+jms_serializer:
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ShippingBundle/Resources/config/app/state_machine.yml]: This should not be here.

Comment (pamil) [inline src/Sylius/Bundle/ShippingBundle/Resources/config/app/state_machine.yml]: 👍

**Comment by michalmarcinkowski** [comment]
Good job! 👍

**Comment by nakashu** [comment]
@pamil why where all migrations removed ??? That seems a bit drastic for upgrading.

**Comment by psyray** [comment]
Agree with @nakashu, remove migrations is a really bad idea for those who want to upgrade from very old sylius version.
And this part was not really discussed her.
I think this part should be quickly reverted.

**Comment by pjedrzejewski** [comment]
@psyray We will publish old migrations in a separate repo probably, because we need to generate new ones and figure out proper upgrade strategy that considers application + sylius migrations together.

**Comment by psyray** [comment]
OK, right, but meanwhile there is no possibility to cleanly migrate old sylius for those who wants.
Maybe you could put a dir named old with old migrations and a README textfile in migrations folder to permit user to manually copy/paste migrations in main folder ?

**Comment by peteward** [comment]
The old migrations didn't work for anyone with a DB that has content because they didn't consider any data - they just did structure. So you would get a same result by just doing a `doctrine:migrations:diff` and then working through your issues.

**Comment by psyray** [comment]
I'm not agree with you.
Some old migrations have logic and move data from a table to another, you can look at those with code migrations or translations (like [Version20160112154949](https://github.com/Sylius/Sylius/blob/v0.18.0/app/migrations/Version20160112154949.php) [Version20151203134947](https://github.com/Sylius/Sylius/blob/v0.18.0/app/migrations/Version20151203134947.php) or [Version20141203091118](https://github.com/Sylius/Sylius/blob/v0.18.0/app/migrations/Version20141203091118.php)).

**Comment by psyray** [comment]
Another example is channel, which insert default channel and update the whole channel tables.
[Version20150202213852](https://github.com/Sylius/Sylius/blob/v0.18.0/app/migrations/Version20150202213852.php)

**Comment by psyray** [comment]
More recent option_value translation
[Version20160203224648](https://github.com/Sylius/Sylius/blob/v0.18.0/app/migrations/Version20160203224648.php)

I'm agree that a lot of migrations was obsolete and was just a `doctrine:migrations:diff`but some are not, and drop the entire migration folder was a mistake.

**Pull request merged**